content/articles/json-server-and-axios/index.md

```diff
@@ -0,0 +1,350 @@
+```bash
+npm install axios
+```
+After the installation, we can now import axios in our application. Inside App.vue, add the following code:
+```js
+<template>
+ <div id="app">
+   <h1>Shopping List</h1>
+ </div>
+ <ul>
+   <li v-for="item of items" :key="item.id">
+     {{item.name}}
+   </li>
+ </ul>
+</template>
+
+<script>
+import axios from 'axios'
+
+export default {
+  name: 'App',
+  data () {
+    return {
+      items: []
+    }
+  },
+  async created (){
+    try {
+      const res = await axios.get(`http://localhost:3000/items`)
+      this.items = res.data;
+    } catch (error) {
+      console.log(error)
+      }
+  },
+}
+</script>
+
+<style>
+#app {
+  text-align: center;
+  color: #2c3e50;
+  margin-top: 60px;
+}
+li {
+  list-style:none;
+}
+</style>
+
+```
+From the above code, we're utilizing the [async and await](https://www.javascripttutorial.net/es-next/javascript-async-await/) keywords, which build on promises and allow you to construct asynchronous code.
+
+### Adding data using `POST`
+You can make `POST`, `PUT`, `PATCH` or `DELETE` requests to the database using `Axios`.
+
+`POST` is a verb (also refered to verb) of HTTP used to create or add data in the database. In our project, let's add POST method to add an item in our shopping list:
+```js
+<template>
+ <div id="app">
+   <h1>Shopping List</h1>
+   <input v-model="itemName"  type="text"><br>
+   <button @click="addItem()">Add Item</button>
+ </div>
+ <ul>
+   <li v-for="item of items" :key="item.id">
+     {{item.name}}
+   </li>
+ </ul>
+</template>
+
+<script>
+import axios from 'axios'
+
+export default {
+  name: 'App',
+  data () {
+    return {
+      items: [],
+      itemName: ''
+    }
+  },
+  async created (){
+    try {
+      const res = await axios.get(`http://localhost:3000/items`)
+      this.items = res.data;
+    } catch (error) {
+      console.log(error)
+      }
+  },
+  methods: {
+    async addItem () {
+        const res = await axios.post(`http://localhost:3000/items`,{name: this.itemName})
+          this.items = [...this.items,res.data]
+          this.itemName = ''
+        }
+  }
+}
+</script>
+
+<style>
+#app {
+  text-align: center;
+  color: #2c3e50;
+}
+li {
+  list-style:none;
+}
+button {
+  margin-top: 5px;
+  background-color: #3498db;
+  border: none;
+  color: #ffffff;
+  padding: 10px 20px;
+  font-size: 14px;
+  cursor: pointer;
+  border-radius: 4px;
+}
+input{
+  margin-top: 5px;
+  padding: 10px 20px;
+  font-size: 14px;
+  border-radius: 4px;
+}
+</style>
+```
+Here, we have added an input area and a button. We have also added a method that is called once you click the button to add the item. JSON-server handles id property by incrementing by 1 automatically. We are then setting the new items to contain every item element, and then add the item to the end of the list.
+
+### Updating and deleting data using `PATCH` and `DELETE` respectively
```

Comment:
This heading seems too long. Consider splitting it into two. Remember to update the table of contents.